lib/pbench/test/functional/server/test_connect.py

```diff
@@ -26,3 +30,26 @@ def test_connect(self, server_client: PbenchServerClient):
         for e in expected:
             assert e in endpoints["api"].keys()
             assert e in endpoints["uri"].keys()
+
+    def test_keycloak(self, server_client: PbenchServerClient):
+        endpoints = server_client.endpoints
+        logger = logging.getLogger("FUNCTIONAL_TEST")
+        oidc_client = OpenIDClient(
+            server_url=endpoints["authentication"]["issuer"],
+            client_id=endpoints["authentication"]["client"],
+            realm_name=endpoints["authentication"]["realm"],
+            client_secret_key=endpoints["authentication"]["secret"],
+            logger=logger,
+        )
+        assert (
+            oidc_client.TOKENINFO_ENDPOINT
+            == "http://localhost:8090/realms/pbench/protocol/openid-connect/token/introspect"
```

Comment:
I would go one step further and use a regex:  while it's true that in our very specific test organization the host is `localhost` (and the port is `8090` and the scheme is `http`), in the general case this won't be necessarily be true, so it would probably be better if we excluded them from the test (or, at least permitted a match which included `https`, a missing or different port, and something else for the hostname).

Reply:
It's not much effort...e.g.,
```
	assert re.fullmatch(
            "https?://[^/]+/realms/[^/]+/protocol/openid-connect/token/introspect",
            oidc_client.TOKENINFO_ENDPOINT,
        )
```
And, this would mean that the test could be used even if the target is on a remote node, or if it uses another NIC, or a FQDN instead of `localhost`, or if it uses a "test realm", etc.

Moreover, this is compatible with @portante's initial suggestion:
```
        prefix = "https?://[^/]+/realms/[^/]+/protocol/openid-connect"
...
        assert re.fullmatch(
            f"{prefix}/token/introspect", oidc_client.TOKENINFO_ENDPOINT
        )
```

Reply:
This test is removed now based on this conversation https://github.com/distributed-system-analysis/pbench/pull/3065#discussion_r1028196282